version/version.go

```diff
@@ -0,0 +1,97 @@
+package version
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	notifier "github.com/similarweb/client-notifier"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	// Version of the release, the value injected by .goreleaser
+	version = `{{.Version}}`
+
+	// Commit hash of the release, the value injected by .goreleaser
+	commit = `{{.Commit}}`
+	// ErrVersionResp response if version was not found in memory
+	ErrVersionResp = errors.New("Version response was not found")
+)
+
+// VersionManagerDescriptor describe the version interface
+type VersionManagerDescriptor interface {
+	Get() (*notifier.Response, error)
+}
+
+// Version struct
+type Version struct {
+	duration        time.Duration
+	params          *notifier.UpdaterParams
+	requestSettings notifier.RequestSetting
+	response        *notifier.Response
+}
+
+// NewVersion creates new instance of version
+func NewVersion(ctx context.Context, duration time.Duration, requestSettings notifier.RequestSetting) *Version {
+
+	params := &notifier.UpdaterParams{
+		Application:  "finala",
+		Organization: "similarweb",
+		Version:      version,
+	}
+
+	version := &Version{
+		params:          params,
+		requestSettings: requestSettings,
+		duration:        duration,
+	}
+
+	response, err := notifier.Get(version.params, version.requestSettings)
+	version.printResults(response, err)
+	version.interval(ctx)
+
+	return version
+}
+
+// interval is a periodic version checker
+func (v *Version) interval(ctx context.Context) {
+	notifier.GetInterval(ctx, v.params, v.duration, v.printResults, v.requestSettings)
+}
+
+// printResults print the notifier response
+func (v *Version) printResults(notifierResponse *notifier.Response, err error) {
+
+	if err != nil {
+		log.WithError(err).Debug(fmt.Sprintf("failed to get Finala latest version"))
```

Comment:
I don't understand how the linter action pass but you need to remove the `fmt.Sprintf` because you don't have variables

Reply:
fixed.